.github/scripts/update_schutzfile.py

```diff
@@ -0,0 +1,55 @@
+#!/bin/python3
+
+import json
+import os
+import re
+import argparse
+
+parser = argparse.ArgumentParser(description="Updates Schutzfile repositories.")
+parser.add_argument(
+    "--suffix",
+    metavar="SUFFIX",
+    type=str,
+    help="date of the rpmrepo snapshots",
+    required=True,
+)
+parser.add_argument(
+    "--repo",
+    metavar="REPO",
+    type=str,
+    help="repository containing Schutzfile",
+    required=True,
+)
+args = parser.parse_args()
+
+repo_files = os.listdir("repo/")
+singletons = []
+# Get a list of all repositories that contain 'singleton'
+for repo_file in repo_files:
+    with open(os.path.join("repo", repo_file), "r") as f:
+        data = json.load(f)
+    if "singleton" in data.keys():
+        singletons.append(data["snapshot-id"])
+
+with open(os.path.join(args.repo, "Schutzfile"), "r") as f:
+    data = json.load(f)
+
+urls = []
+# Replace snapshot SUFFIX in repositories that don't have 'singleton' present
+for key in data.keys():
+    if "repos" in data[key].keys():
+        for repo in data[key]["repos"]:
+            for arch_repos in repo:
+                if arch_repos == "file" or any(
+                    singleton in repo[arch_repos][0]["baseurl"]
+                    for singleton in singletons
+                ):
+                    continue
+                repo[arch_repos][0]["baseurl"] = re.sub(
```

Comment:
We need to iterate over all repos here, otherwise it will update only the first one. (I tested the script locally on composer's Schutzfile and it updated only the first repository for a specific distroarch combination (only for CentOS and RHEL)).

Reply:
Aaah, damn, yeah you're right, thanks for noticing, I've checked the fedoras and then saw a lot of changes so I thought it was fine :disappointed: